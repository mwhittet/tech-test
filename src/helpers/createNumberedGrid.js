import { numbers } from "./createArrayOfNumbers";
import { createNumberedBox } from "./createNumberedBox";

export const createNumberedGrid = array => array.map(createNumberedBox);

export const grid = createNumberedGrid(numbers);
