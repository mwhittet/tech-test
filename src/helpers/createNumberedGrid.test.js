import { createNumberedGrid } from "./createNumberedGrid";

describe("helpers > createNumberedGrid", () => {
  it("returns an array of divs with their own numeric value", () => {
    const array = ["1", "2", "3", "4"];
    const result = createNumberedGrid(array);

    expect(Array.isArray(result));
    expect(result).toEqual(expect.arrayContaining(result));
    expect(result).toMatchSnapshot();
  });
});
