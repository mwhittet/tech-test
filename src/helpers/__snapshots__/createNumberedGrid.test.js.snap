// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`helpers > createNumberedGrid returns an array of divs with their own numeric value 1`] = `
Array [
  <div
    class="box"
  >
    1
  </div>,
  <div
    class="box"
  >
    2
  </div>,
  <div
    class="box"
  >
    3
  </div>,
  <div
    class="box"
  >
    4
  </div>,
]
`;
